monochart/README.md

```diff
@@ -133,10 +132,188 @@ releases:
+              alb.ingress.kubernetes.io/scheme: internet-facing
+              # Health check path has to be adjusted according to your application so load balancer can check if it can 
+              # route traffic to your service.
+              alb.ingress.kubernetes.io/healthcheck-path: /
+```
+
+Then, we have to update `hosts` section. There are 2 notations supported.
+
+1. Legacy. Backward compatibility with Network Load Balancer. 
+   
+   This mode helps to migrate from `nginx` ingress class to `alb`, however, it's not secure at all as the ZScaler
+   configuration is skipped and entire endpoint is exposed to public internet. `hosts` ingress section doesn't require
+   any modifications and can be left as it is. Just add annotations mentioned above and you're good to go.
+
+   ```yaml
+            hosts:
+              # APP_HOST is calculated and provided from CI/CD
+              # It represents the hostname to route for, and the `/` the path to forward requests to
+              '{{ env "APP_HOST" }}': /
+              # Allows us to optionally define extra DNS names to send to this ingress
+              {{- if index .Values "vanityDomain" }}
+              '{{ .Values.vanityDomain }}': /
+              {{- end }}
+   ```
+
+    Live apps examples:
+
+    - [olo-web-api](https://github.com/SpotOnInc/giftcard-service/blob/36addd4fa2d6ad6fa2105089433d7c8e2ec1da7c/deploy/releases/giftcard-api.yaml#L121-L128) 
+    - [dishout-service](https://github.com/SpotOnInc/dishout-service/blob/22547a1da6c1054ec282c192aff630bf84666385/deploy/releases/dishout-service.yaml#L291)
+    - [giftcard service](https://github.com/SpotOnInc/giftcard-service/blob/36addd4fa2d6ad6fa2105089433d7c8e2ec1da7c/deploy/releases/giftcard-api.yaml#L121-L128)
+
+1. Hide everyting behind ZScaler but expose some endpoints.
+
+   This is a preferred way of keeping URLs private because it restricts access by default without extra configuration.
+   The hosts section should look like below:
+
+   ```yaml
+            hosts:
+              '{{ env "APP_HOST" }}': 
+              {{- if index .Values "public_paths" }}
+                {{- range .Values.public_paths }}
+                - '{{ . }}'
+                {{- end }}
+              {{- end }}
+
+              # Allows us to optionally define extra DNS names to send to this ingress.
+              # The $root variable is important because the nested range loop has to refer to top-level .Values 
+              {{- $root := . -}}
+            {{- if index .Values "vanityDomains" }}
+              {{ range $domain := .Values.vanityDomains }}
+              '{{ $domain }}':
+                {{- if index $root.Values "public_paths" }}
+                  {{- range $path := $root.Values.public_paths }}
+                - '{{ . }}'
+                  {{- end }}
+                {{- end }}
+              {{- end }}
+            {{- end }}
```

Comment:
Indents looks improper but for go template it doesn't matter. I try to keep it with indents to not lose track of `{{- <function> }}` `{{ -end }}` pair. It's a compromise between readability and template values placement respectivity.